packages/js/product-editor/src/components/block-slot-fill/block-fill.tsx

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import { Fill } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getName } from './utils/get-name';
+import { BlockFillProps } from './types';
+
+export function BlockFill( { name, clientId, ...props }: BlockFillProps ) {
+	const rootClientId = useSelect(
+		( select ) => {
+			const { getBlockRootClientId } = select( 'core/block-editor' );
+			return getBlockRootClientId( clientId );
```

Comment:
Should we be using `getBlockParentsByBlockName` or maybe `getBlockParents`?
That would only mean that `BlockFill` may not be as generic, and should either be `Section` specific -> `BlockSectionFill`, or add something for the block name as well?

Reply:
Well I decided to do it that way so we can use this logic not only for the section block but also in any other block.